R/get_bw_matrix.R

```diff
@@ -40,7 +43,8 @@ get_bw_matrix.BigWigFileList <- function(bw, regions,
 #' @importFrom GenomeInfoDb seqlevels seqlevels<-
 #' @importFrom rtracklayer summary
 get_bw_matrix.BigWigFile <- function(bw, regions,
-                          type = c("mean", "min", "max", "coverage", "sd")){
+                          type = c("mean", "min", "max", "coverage", "sd"),
+                          by = NULL){
```

Comment:
added `by = NULL` that will give option to out a list of matrices split by regions mcol variable 
TO DO
- [ ] fully test this feature - what breaks it?
- [ ] add check that `by` is a valid mcol of regions provide useful error

Reply:
>add check that by is a valid mcol of regions provide useful error

Think this untested code chunk will do it.
```
if (!is.null(by)){
  if (!(by %in% names(mcols(regions))) {
     msg <- paste0("'by' value:", by, " is not a valid column name of regions".)
     stop(msg, .call = FALSE)
  }
}

```

If we wanted to get _really_ fancy, we could use `stringdist` to find the `mcol` name with the lowest Levenstein edit distance to the value passed to `by` to suggest a "did you mean?".